src/VotingMultipliers.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {IVotingMultipliers, IMultiplier} from "src/interfaces/IVotingMultipliers.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+/// @title VotingMultipliers
+/// @notice A contract for managing voting multipliers
+/// @dev Implements IVotingMultipliers interface
+contract VotingMultipliers is OwnableUpgradeable, IVotingMultipliers {
+    /// @notice Array of allowlisted multiplier contracts
+    IMultiplier[] public allowlistedMultipliers;
+
+    /// @notice Calculates the total multiplier for a given _user
+    /// @param _user The address of the _user
+    /// @return The total multiplier value for the _user
+    function getTotalMultipliers(address _user) public view returns (uint256) {
```

Comment:
Assuming we keeping this as a quality of life function so we can avoid the double query `getValidMultiplierIndexes(address)` -> `getTotalMultipliers(address, uint256[])`? If so, I think this should be moved to be closer to `getTotalMultipliers(address, uint256[])` so the overload is obvious. Maybe word adding a note in the natspec comments with a @dev tag

Reply:
This is more for frontend and for testing. Moved closer and added dev tag with feaeefd